sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -282,21 +283,28 @@ private[sql] class ParquetRelation2(
         val cacheMetadata = useMetadataCache
 
         @transient val cachedStatuses = inputFiles.map { f =>
-          // In order to encode the authority of a Path containing special characters such as /,
-          // we need to use the string returned by the URI of the path to create a new Path.
-          val pathWithAuthority = new Path(f.getPath.toUri.toString)
```

Comment:
Just a note: When we create a Path from a string, we actually internally create a URI.